judo-runtime-core-dao-core/src/main/java/hu/blackbelt/judo/runtime/core/dao/core/processors/InsertPayloadDaoProcessor.java

```diff
@@ -166,6 +166,25 @@ Collection<Statement<ID>> collectStatements(EClass mappedTransferObjectType,
                                 getTransferObjectValueAsEntityValueFromPayload(payload, a.getKey(), a.getValue()))
         );
 
+        references.stream()
+                .filter(r -> getAsmUtils().getMappedReference(r).isPresent()
+                        && getAsmUtils().getMappedReference(r).orElseThrow().isContainment()
+                        && payload.containsKey(r.getName())
+                        && payload.get(r.getName()) != null)
+                .collect(toReferencePayloadMapOfPayloadCollection(payload))
+                .entrySet().stream()
+                .forEach(entry -> entry.getValue().stream()
+                        .forEach( payloadStm -> {
+                                    if (payloadStm.get(getIdentifierProvider().getName()) != null) {
+                                        payloadStm.remove(getIdentifierProvider().getName());
+                                        payloadStm.remove(ENTITY_TYPE_KEY);
+                                        payloadStm.remove(VERSION);
+                                    }
```

Comment:
It may seem a bit "hacky," but I believe that simply removing every technical entry that starts with "__" in their keys would do the job.